controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -162,12 +196,78 @@ func (r *StorageConsumerUpgradeReconciler) Reconcile(ctx context.Context, reques
 		consumerConfigMap.Data = data
 
 		if err := r.Client.Create(ctx, consumerConfigMap); err != nil {
-			return reconcile.Result{}, err
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *StorageConsumerUpgradeReconciler) reconcileStorageRequest(
+	ctx context.Context,
+	storageCluster *ocsv1.StorageCluster,
+	storageConsumer *ocsv1alpha1.StorageConsumer,
+) error {
+
+	rbdClaimName := util.GenerateNameForCephBlockPoolStorageClass(storageCluster)
+	rbdStorageRequestName := getStorageRequestName(string(storageConsumer.UID), rbdClaimName)
+	rbdStorageRequestMd5Sum := md5.Sum([]byte(rbdStorageRequestName))
+	rnsName := fmt.Sprintf("cephradosnamespace-%s", hex.EncodeToString(rbdStorageRequestMd5Sum[:16]))
+
+	// Removing the onwerRef from rns to preserve it after deletion the StorageRequest
+	rns := &metav1.PartialObjectMetadata{}
+	rns.SetGroupVersionKind(rookCephv1.SchemeGroupVersion.WithKind("CephBlockPoolRadosNamespace"))
+	rns.Name = rnsName
+	rns.Namespace = storageConsumer.Namespace
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(rns), rns); client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if rns.UID != "" && removeStorageRequestAsOwner(rns) {
+		if err := r.Client.Update(ctx, rns); client.IgnoreNotFound(err) != nil {
```

Comment:
either use a full object or do a patch here as you need entire view for performing update.

Reply:
Updated to use patch